lua/entities/gmod_wire_expression2/core/weapon.lua

```diff
@@ -78,3 +134,114 @@ e2function string entity:tool()
+	if not target:IsPlayer() then return self:throw("Expected a Player but got Entity", false) end
+	if not CanTool(self.entity.player, target, "wire_expression2") then return self:throw("You cannot target this player", false) end
+	if not list.HasEntry("Weapon", classname) then return self:throw("Expected a weapon class, but the weapon was not found", false) end
+	if hook.Run('PlayerGiveSWEP', target, classname, list.Get('Weapon')[classname]) == false then
+		return self:throw("The server blocked the weapon from being given", false)
+	end
+
+	return true
+end
+
+e2function entity entity:giveWeapon(string classname)
+	if not checkGive(self, this, classname) then return NULL end
+
+	return this:Give(classname)
+end
+
+e2function entity entity:giveWeapon(string classname, noAmmo)
+	if not checkGive(self, this, classname) then return NULL end
+
+	return this:Give(classname, noAmmo ~= 0)
+end
+
+e2function void entity:selectWeapon(string classname)
+	if not IsValid(this) then return self:throw("Invalid entity!", nil) end
+	if not this:IsPlayer() then return self:throw("Expected a Player but got Entity", nil) end
+	if not CanTool(self.entity.player, this, "wire_expression2") then return self:throw("You cannot target this player", nil) end
+
+	this:SelectWeapon(classname)
+end
+
+e2function number entity:giveAmmo(amount, string type)
+	if not giveAmmoCVar:GetBool() then return self:throw("The server has disabled giving ammo", 0) end
+	if not IsValid(this) then return self:throw("Invalid entity!", 0) end
+	if not this:IsPlayer() then return self:throw("Expected a Player but got Entity", 0) end
+	if not CanTool(self.entity.player, this, "wire_expression2") then return self:throw("You cannot target this player", 0) end
+
+	return this:GiveAmmo(amount, type)
+end
+
+e2function number entity:giveAmmo(amount, string type, hidePopUp)
+	if not giveAmmoCVar:GetBool() then return self:throw("The server has disabled giving ammo", 0) end
+	if not IsValid(this) then return self:throw("Invalid entity!", 0) end
+	if not this:IsPlayer() then return self:throw("Expected a Player but got Entity", 0) end
+	if not CanTool(self.entity.player, this, "wire_expression2") then return self:throw("You cannot target this player", 0) end
+
+	return this:GiveAmmo(amount, type, hidePopUp ~= 0)
+end 
+
+e2function void entity:setAmmo(ammoCount, string type)
+	if not giveAmmoCVar:GetBool() then return self:throw("The server has disabled giving ammo", 0) end
+	if not IsValid(this) then return self:throw("Invalid entity!", nil) end
+	if not this:IsPlayer() then return self:throw("Expected a Player but got Entity", nil) end
+	if not CanTool(self.entity.player, this, "wire_expression2") then return self:throw("You cannot target this player", nil) end
+
+	this:SetAmmo(ammoCount, type)
+end
+
+e2function void entity:removeAmmo(ammoCount, string type)
+	if not IsValid(this) then return self:throw("Invalid entity!", nil) end
```

Comment:
Second thought, probably do want to check `setAmmoCvar` on the remove* functions. Could imagine it going quite bad when a server gives people CPPI access and some guy just forces everyones ammo to 0.